src/XMakeTasks/FileIO/WriteLinesToFile.cs

```diff
@@ -110,29 +111,33 @@ public override bool Execute()
                         }
                         else
                         {
-                            // Passing a null encoding, or Encoding.Default, to WriteAllText or AppendAllText
-                            // is not the same as calling the overload that does not take encoding!
-                            // Encoding.Default is based on the current codepage, the overload without encoding is UTF8-without-BOM.
-                            if (encode == null)
+                            string contentsAsString = null;
+
+                            // When WriteOnlyWhenDifferent is set, read the file and if they're the same return.
+                            if (WriteOnlyWhenDifferent && System.IO.File.Exists(File.ItemSpec))
                             {
-                                System.IO.File.WriteAllText(File.ItemSpec, buffer.ToString());
+                                var existingContents = System.IO.File.ReadAllText(File.ItemSpec);
```

Comment:
What happens when `ReadAllText` throws?  The use of `File.Exists` above doesn't give an guarantees this will succeed.